File: CreationalPattern/FactoryPattern/Java/Plants/SunFlower.java

package Plants;

import Interface.Plant;

public class SunFlower implements Plant {
    // Var parameters
    private String name;
    private int health;
    private int price;
    // TODO parameters

    // Constructor
    public SunFlower() {
        super();
        this.name = "SunFlower";
        this.health = 100;
        this.price = 50;
    }

    public SunFlower(String name, int health, int price) {
        super();
        this.name = name;
        this.health = health;
        this.price = price;
    }

    // Override method show
    @Override
    public void show() {
        System.out.println("Name: " + name);
        System.out.println("Health: " + health);
        System.out.println("Price: " + price);
    }

    // Getter and Setter 
    // TODO add ...
}

File: CreationalPattern/FactoryPattern/Python/Plants/SunFlower.py

from Interface import Plant

class SunFlower(Plant):
    """ 
        SunFlower class
    """
    def __init__(self, *args, **kwargs):
        if args:
            super().__init__(args[0])
            self.price = args[1]
            self.health = args[2]
        elif kwargs:
            super().__init__(kwargs.get('name','SunFlower'))
            self.price = kwargs.get('price', 50)
            self.health = kwargs.get('health', 100)
        else:
            super().__init__('SunFlower')
            self.price = 50
            self.health = 100
    
    def show(self):
        print("Name: " + self.name 
                + " Price: " + str(self.price) 
                + " Health: " + str(self.health))
